### ISSUE TITLE
CV upload sometimes garbles the whitespace

### ISSUE THREAD

**Issue by lilsweetcaligula** [DevvJobs/candidates-web#45]
The issue occurs occasionally. Some parts of the PDF are logged correctly, whereas some others are missing whitespace, e.g. `hello world` becomes `helloworld`. This can be problematic for skill parsing.

I'll see if I can look into this problem, but I think I remember ![seeing](https://gitlab.com/autokent/pdf-parse/blob/master/lib/pdf-parse.js#L22) a mention of this behavior and the subsequent fix in the `pdf-parse` package source code. Something to do with `transforms`. I had some problems setting up the package and so opted out of using it for the time-being.

**Comment by ivan-kolmychek-devv** [comment]
Caught that today as well during testing of #44.

My impression is that it can mess up detection quite significantly.

Any suggestions on we can work around it?

**Comment by lilsweetcaligula** [comment]
@ivan-kolmychek-devv Yeah, I believe the source code to the `pdf-parse` has the answer. (Sorry, the link was broken, now fixed).

I believe the reason I could not use `pdf-parse` had something to do with the `fs` module not being available. But I feel that enabling support of `fs` on front-end is a serious security vulnerability, so we'll probably have to "adapt" the solution.

I'll try checking out alternative packages.

**Comment by ivan-kolmychek-devv** [comment]
@lilsweetcaligula [wrote](https://github.com/DevvJobs/candidates-web/pull/44#issuecomment-476592313) that he does not have time anymore for this issue.

If anyone wants to pick it up or contribute in any other way - go ahead. :+1:

**Comment by lilsweetcaligula** [comment]
Basically, the whitespace issue seems to occur with the PDF packages as well. It probably has to do with the way `pdf.js` interprets PDF content as strings, but can probably be worked around.

**Comment by ivan-kolmychek-devv** [comment]
Deployed to https://dev.devv.jobs.
